ptvsd/wrapper.py

```diff
@@ -777,6 +777,13 @@ def on_evaluate(self, request, args):
         xml = untangle.parse(resp_args).xml
         xvar = xml.var
 
+        if xvar['isErrorOnEval']=='True':
+            self.send_response(
+                request,
+                result=None,
```

Comment:
What happens if the user enters an invalid expression in Watch - it would seem that they don't get any information about what the error is like?

For this particular bug (#65), it seems that the root cause is improper detection of what constitutes a hover expression in VS (i.e. it shouldn't be sending things like `def` for evaluation in the first place) - this is what `DetectSideEffectsWalker` and `GetDataTipText` in PTVS is about. If possible, we should just figure out how to hook that up into the VSCode debug adapter.  

For cases where there genuinely is an error, we do want to report an error (like the old debugger does in Watch if you put an invalid expression there). The only thing missing is the appropriate error icon (instead of the usual variable icon) on the result - I think this was in `VariablePresentationHint` in the protocol.

Reply:
I will check with the adapter host folks about improper detection of hover expressions. But to isolate hover cases, how about we check the `EvaluateArguments.enumDescriptions`, and only return the `None` result for those cases?

That way Watch will still get the error. But i did not see a way to provide error in `VariablePresentationHint`. I will check with the adapter host folks.

Reply:
Indeed, it doesn't seem to be in the procotol. Which is strange, since it's mostly just a mapping of VS constants for variable types, and that does include errors. Must be an oversight. 

`enumDescriptions` sounds like a good way to go - then we'll have defense in depth on ptvsd side even after we fix the hover expression boundaries.